### ISSUE TITLE
Hiding and showing projects displayed on my My StaffPlan

### ISSUE THREAD

**Issue by ericbenwa** [goinvo/staffplan-next-app#437]
New functionality.
This has nothing to do with archiving – different topic.

Purpose for the functionality:
I completed a project and no longer want it to display on My StaffPlan as it's no longer relevant to me... and I don't want to visually distract from the other projects I'm focused on... aka clean up my view to the important/current projects.

Auth'd user can hide/show projects they are connected to from displaying on their My StaffPlan.
Hiding a project from My StaffPlan has no impact on anyone else's view of that project.

Hiding a project from My StaffPlan will not hide hours from the totals in the top (bar chart).

I cannot hide/show projects on anyone else's My StaffPlan.

See design for more details.
Might be helpful to jump on a call to walk through it (since I haven't typed up all the details here) before any engineering begins.
Let me know!

![hide-show-my-staffplan](https://github.com/user-attachments/assets/808ad308-e674-4b3c-ae1e-899b1ea16f7e)

**Comment by zhukovdigital** [comment]
@fermion Perhaps we need to add new functionality to the backend first? Or is there already something for this task?

**Comment by fermion** [comment]
@zhukovdigital yep, agreed that what we currently have for metadata for an assignment doesn't really capture this adequately. Let me think about how this might be modeled on the backend.

@ericbenwa re: prioritization, how important is this?

**Comment by ericbenwa** [comment]
@fermion Not urgent at the moment. 
Will eventually grow over time as people complete more projects and it'll become a list of projects they are no longer working / no longer need to see.
But we can hold on this for the time being.

**Comment by ericbenwa** [comment]
@fermion I was off on assessing this tasks urgency.
Could we prioritize this for next week?

**Comment by fermion** [comment]
> @fermion I was off on assessing this tasks urgency. Could we prioritize this for next week?

Too late, you already deprioritized it. No take backsies!

_puts on professional hat_

Some questions first:

> Hiding a project from My StaffPlan will not hide hours from the totals in the top (bar chart).

This feels like it will be an endless source of confusion unless we think about communicating clearly in the UI where the hours that aren't seen are coming from. Maybe it's moot since this is likely to be used in the scenario where a project is still active, but I'm not a part of it any longer. Admittedly I did skim this quickly so forgive me if you'd addressed it.

> The default should be to hide.
> But if I change to show, that choice should persist for my session.
> So if I navigate away and back to My StaffPlan, I still see w/e choice I've made.

I'm reading this as "if I've hidden an assignment from My StaffPlan that I then decide to unhide, then it should stay unhidden until I hide it again". Is that what you're saying?

> Extra credit: If a project has been hidden from a My StaffPlan... and you or someone else edits your future hours, then the hidden project flips to being not hidden / now shown back in that persons My StaffPlan.

Challenge accepted!
